### PR TITLE
Add Web Action Shortcuts to CLI

### PR DESCRIPTION
**Pull request by dubeejw**
- Adds --web for the enablement of web actions
  - Usuage: `--web default|raw|none`
- Tests for new flag
- Refactoring of action command

### CODE REVIEW

File: tests/src/test/scala/common/Wsk.scala [hunk 1]
```diff
@@ -922,6 +928,10 @@ sealed trait RunWskCmd extends Matchers {
         jsonStr.substring(jsonStr.indexOf("\n") + 1).parseJson.asJsObject // Skip optional status line before parsing
     }
 
+    def parseJsonValueString(jsonStr: String): JsValue = {
```
Comment (rabbah) [inline tests/src/test/scala/common/Wsk.scala]: dont we have this method somewhere else? vaguely recall you added something like it recently.

File: tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala [hunk 2]
```diff
@@ -114,4 +114,127 @@ class WskWebActionsTests
...
+        (wp, assetHelper) =>
+            val name = "webaction"
+            val file = Some(TestUtils.getTestActionFilename("echo.js"))
+
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) =>
+                    action.create(name, file, web = Some(true))
+            }
+
+            val stdout = wsk.action.get(name, fieldFilter = Some("annotations")).stdout
```
Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: think you can collapse the 3 repetitions with a Seq(true, false, true).foreach ...

**Comment by dubeejw** [comment]
PG4 69, approved for first commit.

File: tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala [hunk 3]
```diff
@@ -114,4 +114,63 @@ class WskWebActionsTests
...
+            }
+    }
+
+    it should "ensure --raw and --no-raw set the proper annotations" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "webaction"
+            val file = Some(TestUtils.getTestActionFilename("echo.js"))
+            var updateAction = false
+
+            try {
```
Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: the asset cleaner does this for you.

**Comment by dubeejw** [comment]
PG1 1274, PG4 71.

File: tools/cli/go-whisk-cli/wski18n/resources/en_US.all.json [hunk 4]
```diff
@@ -1366,5 +1366,17 @@
...
+  {
+    "id": "treat ACTION as a web action",
+    "translation": "treat ACTION as a web action"
+  },
+  {
+    "id": "treat ACTION as a raw HTTP web action",
+    "translation": "treat ACTION as a raw HTTP web action"
+  },
+  {
+    "id": "treat ACTION as a non-web action",
```
Comment (mdeuser) [inline tools/cli/go-whisk-cli/wski18n/resources/en_US.all.json]: Is there a better label for a "non-web action"

**Comment by dubeejw** [comment]
@mdeuser, when creating or updating an action, the `--web` flag adds annotations to the corresponding action. Essentially `--web` is a shortcut so that users do not have to type `-a web-export true -a final true` during action create/update. When `--web` is not specified, annotations are not modified automatically by the CLI. In contrast, `--no-web` automatically sets `-a web-export false -a final false`. The other flags, `--raw` and `--no-raw`, do the same as the web commands except modifications are done to an annotation called `raw-http`.

**Comment by dubeejw** [comment]
@mdeuser, currently the backend will delete all existing annotations when a new annotation is added to an action. If we automatically set web action annotations to false whenever a call to `action update` is made without `--raw|--web` we will delete the existing annotations on an action.

**Comment by dubeejw** [comment]
@mdeuser, by having `--no-web|--no-raw` flags we allow users to update actions without having to specify their annotations every time.

**Comment by dubeejw** [comment]
PG4 75.

**Comment by mdeuser** [comment]
Instead of `--web (yes|true)|raw|(no|false)`, how about `--web none|raw|default` ?

**Comment by dubeejw** [comment]
PG4 76.

File: tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala [hunk 5]
```diff
@@ -196,4 +197,88 @@ abstract class WskWebActionsTests
             response.statusCode shouldBe 406
             response.body().asString() should include("Resource representation is only available with these Content-Types:\\ntext/html")
     }
+
+    it should "ensure --web and --no-web flags set the proper annotations" in withAssetCleaner(wskprops) {
```
Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: test name needs an update.

File: tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala [hunk 6]
```diff
@@ -196,4 +196,71 @@ abstract class WskWebActionsTests
             response.statusCode shouldBe 406
             response.body().asString() should include("Resource representation is only available with these Content-Types:\\ntext/html")
     }
```
Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: this is now addressed by consolidating the flags.

**Comment by rabbah** [comment]
Dropped last commit, moved tests and consolidates test cases.
Merged as https://github.com/openwhisk/openwhisk/commit/59e41fbb465c3df75519357b90642910d743aca1.

**Pull request closed without merging**